app/datatables/datatable_base.rb

```diff
@@ -120,4 +123,8 @@ def select_query
   def count_query
     raise 'Must implement in subclass'
   end
+
+  def special_filters
+    raise 'Must implement in subclass'
```

Comment:
I don't know which is better, totally up to you, but you could have this return an empty array as the default behavior and then allow the two subclasses to override it with the parameters they care about. Perhaps less explicit, but also less duplicate methods. Both ways have their pros and cons.